### PR TITLE
[Refactor] reduce column info fields and group reader params

### PR DESCRIPTION
**Pull request by dirtysalt**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/formats/parquet/group_reader.h [hunk 1]
```diff
@@ -40,21 +40,22 @@ namespace starrocks::parquet {
...
+        int32_t idx_in_parquet;
 
         // column type in parquet file
-        tparquet::Type::type col_type_in_parquet;
+        tparquet::Type::type type_in_parquet;
 
-        // column type in chunk
-        TypeDescriptor col_type_in_chunk;
+        // column index in chunk
+        int32_t idx_in_chunk;
```
Comment (imay) [inline be/src/formats/parquet/group_reader.h:49]: how about giving all fields a default value of -1?

Comment (dirtysalt) [reply be/src/formats/parquet/group_reader.h:49]: OK.

but this field  will be removed in next pr. in this commit https://github.com/StarRocks/starrocks/pull/42087/commits/d54f396f7a2a8ace966077141b71e614f82ff952

File: be/src/formats/parquet/group_reader.cpp [hunk 2]
```diff
@@ -385,7 +383,7 @@ void GroupReader::_init_read_chunk() {
     const auto& slots = _param.tuple_desc->slots();
     std::vector<SlotDescriptor*> read_slots;
     for (const auto& column : _param.read_cols) {
-        int chunk_index = column.col_idx_in_chunk;
+        int chunk_index = column.idx_in_chunk;
         read_slots.emplace_back(slots[chunk_index]);
```
Comment (zombee0) [inline be/src/formats/parquet/group_reader.cpp:387]: I think this is the final reason, idx_in_chunk use both tuple_desc->slots() and delete_column_tuple_desc->slots(), which lead to oversize.

Comment (dirtysalt) [reply be/src/formats/parquet/group_reader.cpp:387]: yes, see this pr to fix https://github.com/StarRocks/starrocks/pull/42087

Comment (dirtysalt) [reply be/src/formats/parquet/group_reader.cpp:387]: so we'd better not to use `idx_in_chunk`, but to use slot desc, not matter it's from tuple desc or delete column tuple desc.

File: be/src/exec/hdfs_scanner.h [hunk 3]
```diff
@@ -213,16 +213,16 @@ struct HdfsScannerParams {
 
 struct HdfsScannerContext {
     struct ColumnInfo {
-        int col_idx;
-        TypeDescriptor col_type;
-        SlotId slot_id;
-        std::string col_name;
+        int idx_in_chunk;
```
Comment (stephen-shelby) [inline be/src/exec/hdfs_scanner.h:216]: It seems `idx_in_chunk` has not been used, can we delete it?

**Review by zombee0 — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

File: be/src/exec/hdfs_scanner.h (hunk 3 above)
Comment (dirtysalt) [reply be/src/exec/hdfs_scanner.h:216]: fixed in next pr.

**Review by murphyatwork — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by dirtysalt** [comment]
@mergify backport branch-3.2

**Comment by dirtysalt** [comment]
@mergify backport branch-3.1

**Pull request merged**